addons/medical_damage/functions/fnc_determineIfFatal.sqf

```diff
@@ -0,0 +1,60 @@
+#include "script_component.hpp"
+/*
+ * Author: PabstMirror, Glowbal
+ * Determines if damage is fatal
+ *
+ * Arguments:
+ * 0: The Unit <OBJECT>
+ * 1: Part No <NUMBER>
+ * 2: Damage Array - QGVAR(medical,bodyPartDamage) <ARRAY>
+ * 3: New Damage <NUMBER>
+ *
+ * ReturnValue:
+ * Was Fatal <BOOL>
+ *
+ * Example:
+ * [player, 0, 1.4, 0.7] call ace_medical_damage_fnc_determineIfFatal
+ *
+ * Public: No
+ */
+
+params ["_unit", "_part", "_bodyPartDamage", "_woundDamage"];
+
+if (_part > 1) exitWith { false };
+
+scopeName "main";
+
+if (EGVAR(medical,fatalDamageSource) in [0, 2]) then {
+    // Emulate damage to vital organs - Original rewrite logic, only powerfull headshots or random torso shots
+    if (_part == 0 && {_woundDamage >= HEAD_DAMAGE_THRESHOLD}) exitWith {
+        // Fatal damage to the head is guaranteed death
+        TRACE_1("determineIfFatal: lethal headshot",_woundDamage);
+        true breakOut "main";
+    };
+    if (_part == 1 && {_woundDamage >= ORGAN_DAMAGE_THRESHOLD} && {random 1 < HEART_HIT_CHANCE}) exitWith {
+        // Fatal damage to torso has various results based on organ hit - Heart shot is lethal
+        TRACE_1("determineIfFatal: lethal heartshot",_woundDamage);
+        true breakOut "main";
+    };
+};
+if (EGVAR(medical,fatalDamageSource) in [1, 2]) then {
+    // Sum of trauma to critical areas can be fatal (e.g. many small hits)
+    private _damageThreshold = if (isPlayer _unit) then { EGVAR(medical,playerDamageThreshold) } else { EGVAR(medical,AIDamageThreshold) };
+    private _headThreashhold = 1.25 * _damageThreshold;
+    private _bodyThreashhold = 1.5 * _damageThreshold;
+
+    _bodyPartDamage params ["_headDamage", "_bodyDamage"];
+
+    private _chanceFatal = 0;
+    if (_headDamage > _headThreashhold) then { _chanceFatal = _chanceFatal + 0.95 * (_headDamage - _headThreashhold); };
+    if (_bodyDamage > _bodyThreashhold) then { _chanceFatal = _chanceFatal + 0.75 * (_bodyDamage - _bodyThreashhold); };
```

Comment:
I think the parameters should be in a config somehwere and I think this probability function should be displayed differently, with easer tweakable parameters. It should be a cumulative distribution function where you set some characteristic points instead of piecewise linear functions.

Reply:
What would that look like in sqf?